core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +120,19 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  /**
+   * traverses through the table's snapshots and finds the {@link Table}'s
+   * Snapshot that is committed after the specified snapshotId
+   * @return null if passed in snapshot expired, else the snapshot after the passed in snapshot
```

Comment:
I don't think this quite captures what the method is doing because it doesn't mention that this is with respect to the table's current state. How about this?

> Traverses the history of the table's current snapshot and finds the snapshot with the given snapshot id as its parent.

I think this should also document the behavior when:
* The given snapshot ID is not found in the table (`IllegalArgumentException`?)
* The given snapshot ID is not an ancestor of the current table state (`IllegalStateException`?)

Nit: We also prefer to use sentence case for documentation.

Reply:
fixed this.